core/src/main/java/hudson/triggers/SCMTrigger.java

```diff
@@ -549,6 +553,16 @@ public void run() {
             if (job == null) {
                 return;
             }
+            // we can pre-emtively check the SCMPollingDecisionHandler instances here
+            // note that job().poll(listener) should also check this
+            for (SCMPollingDecisionHandler handler : SCMPollingDecisionHandler.all()) {
+                if (!handler.shouldPoll(job)) {
+                    LOGGER.log(Level.FINE, "Skipping polling for {0} due to veto from {1}",
```

Comment:
should this not go in the listener, so that it is available in the SCM polling log?